### DISCUSSION TITLE
xit! for Vim

### DISCUSSION THREAD

**Discussion by ryanolsonx** [jotaen/xit#23] [Show & tell]
I'm developing a syntax highlighter for xit in Vim. https://github.com/ryanolsonx/vim-xit

If you'd like to contribute, that would be great. Right now, it supports some of the things. Enough to make it look like a TODO list. Not a full spec compliant implementation (yet!).
<img width="688" alt="Screen Shot 2022-04-04 at 9 57 20 AM" src="https://user-images.githubusercontent.com/238929/161584010-be123be8-02fc-48f9-9d2a-d4aa96eb873a.png">

**Comment by jotaen** [comment]
Awesome that you’ve started this! I’m also curious to see how the implementation will compare to the [nano highlighting](https://github.com/jotaen/xit/discussions/21), as I’m not familiar with the internals of neither editor.

You apparently have already found the (half-baked) test file! Let me know if or when you’d like to share this on [the website](https://xit.jotaen.net/). I personally don’t mind whether linked projects are work-in-progress or not, as long as you’re good with it.

**Reply by ryanolsonx** [reply]
It'll be similar, but Vim does have support for look-ahead and look-behind in it's syntax regex implementation. So that'll be good.

Yeah, it's been good to test against ;) You can 100% put it on the website, if you'd like. I'm going to keep working on it. For sure a work-in-progress though.

**Reply by jotaen** [reply]
It’s added 🎉

**Comment by jotaen** [comment]
Just fyi, for improved test data: https://github.com/jotaen/xit/issues/22

**Comment by mrzool** [comment]
@ryanolsonx Not even two years later, I'm getting a 404 following [the link to your repo](https://github.com/ryanolsonx/vim-xit) you posted. What happened? 🤔

CC @jotaen that means that the first link to vim-xit on [xit's homepage](https://xit.jotaen.net) is broken right now.

**Reply by jotaen** [reply]
Thanks for the info, I’ve removed that Vim link, and the Emacs one too.

Please note that all these tools have been submitted and are maintained on a voluntary basis.

**Reply by ryanolsonx** [reply]
They’re back

**Reply by mrzool** [reply]
@jotaen 

> Please note that all these tools have been submitted and are maintained on a voluntary basis.

Sure thing, it's just unusual to land on a 404, especially after such a short time. Typically you'd come across an unmaintained or archived repository instead. That's why I assumed it was an error of some sort.

Anyway now the repo seems to be back 🙌 thanks @ryanolsonx

**Reply by jotaen** [reply]
> They’re back

Ok cool, I’ve restored the links.
